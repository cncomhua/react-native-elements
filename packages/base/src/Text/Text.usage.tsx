import { info, usage, Stack } from '@rneui/doc-gen';
import React from 'react';
import { Text } from '..';

info(
  'CheckBoxes allow users to complete tasks that involve making choices such as selecting options, or switching settings - On or Off. It provides a clear visual of either a true or false choice.'
);

usage('Icon', '', () => (
  <Stack>
    <Text h1>Heading 1</Text>
    <Text h2>Heading 2</Text>
    <Text h3>Heading 3</Text>
    <Text h4>Heading 4</Text>
  </Stack>
));
