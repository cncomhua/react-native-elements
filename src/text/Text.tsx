import React from 'react';
import {
  Text,
  StyleSheet,
  Platform,
  TextProps as TextProperties,
  TextStyle,
  StyleProp,
} from 'react-native';
import { fonts, withTheme } from '../config';
import { patchWebProps, RneFunctionComponent } from '../helpers';
import normalize from '../helpers/normalizeText';

type VariantStyle = StyleProp<TextStyle> & {
  h1?: StyleProp<TextStyle>;
  h2?: StyleProp<TextStyle>;
  h3?: StyleProp<TextStyle>;
  h4?: StyleProp<TextStyle>;
};
export type TextProps = TextProperties & {
  style?: VariantStyle;
  variant?: 'h1' | 'h2' | 'h3' | 'h4';

  // To be deprecated
  h1?: boolean;
  h2?: boolean;
  h3?: boolean;
  h4?: boolean;
  h1Style?: StyleProp<TextStyle>;
  h2Style?: StyleProp<TextStyle>;
  h3Style?: StyleProp<TextStyle>;
  h4Style?: StyleProp<TextStyle>;
};

const TextElement: RneFunctionComponent<TextProps> = ({
  style,
  variant,
  theme,
  children = '',

  // To be deprecated
  h1,
  h2,
  h3,
  h4,
  h1Style,
  h2Style,
  h3Style,
  h4Style,
  ...rest
}) => {
  return (
    <Text
      accessibilityRole="text"
      style={StyleSheet.flatten([
        {
          ...Platform.select({
            android: {
              ...(fonts.android.regular as TextStyle),
            },
          }),
          color: theme?.colors?.black,
        },
        style as StyleProp<TextStyle>,
        (variant || h1 || h2 || h3 || h4) && (styles.bold as TextStyle),
        (variant === 'h1' || h1) &&
          StyleSheet.flatten([{ fontSize: normalize(40) }, style?.h1]),
        (variant === 'h2' || h2) &&
          StyleSheet.flatten([{ fontSize: normalize(34) }, style?.h2]),
        (variant === 'h3' || h3) &&
          StyleSheet.flatten([{ fontSize: normalize(28) }, style?.h3]),
        (variant === 'h4' || h4) &&
          StyleSheet.flatten([{ fontSize: normalize(22) }, style?.h4]),
      ])}
      {...patchWebProps(rest)}
    >
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  bold: {
    ...Platform.select({
      android: {
        ...(fonts.android.bold as TextStyle),
      },
    }),
  },
});

export { TextElement };
export default withTheme(TextElement, 'Text');
