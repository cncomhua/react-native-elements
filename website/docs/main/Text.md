---


id: text


title: Text


slug: /text

---



import Usage from './usage/Text/Text.md'



Text displays words and characters at various sizes.



## Usage


<Usage />

---


## Props

### Text




## Reference

### Text

