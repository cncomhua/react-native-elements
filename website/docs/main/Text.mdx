---
id: text

title: Text

slug: /text
---

import Usage from './usage/Text/Text.mdx';

import Props from './props/Text.mdx';

import { Text } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

Text displays words and characters of various sizes.

## Usage

<Usage />

## Props

<Props />
