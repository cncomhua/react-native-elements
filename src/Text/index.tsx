import { Text, TextProps } from './Text';
import { withTheme } from '../config';

export { Text };
export type { TextProps };
/**
 * RNE Text
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/text)
 */
export default withTheme(Text, 'Text');
