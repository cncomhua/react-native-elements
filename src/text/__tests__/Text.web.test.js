import React from 'react';
import { TouchableHighlight, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import Text from '../Text';

const props = {
  'no props': {
    component: Text,
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  h1: {
    component: Text,
    props: { h1: true },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  h2: {
    component: Text,
    props: { h2: true },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  h3: {
    component: Text,
    props: { h3: true },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  h4: {
    component: Text,
    props: { h4: true },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  fontFamily: {
    component: Text,
    props: { fontFamily: 'Courier New' },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  style: {
    component: Text,
    props: { style: { color: '#071' } },
    children: 'Text',
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Misc: { sectionComponents: { Text: { samples } } },
  },
  options
);

describe('Text Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Text />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have font size of 40 when h1', () => {
    const component = shallow(<Text h1 />);

    const styles = component.props().style;
    let fontSizeStyle;
    for (let i = 0; i < styles.length; i++) {
      if (styles[i] && styles[i].hasOwnProperty('fontSize')) {
        fontSizeStyle = styles[i].fontSize;
      }
    }

    expect(fontSizeStyle).toBe(40);
  });

  it('should have font size of 34 when h2', () => {
    const component = shallow(<Text h2 />);

    const styles = component.props().style;
    let fontSizeStyle;
    for (let i = 0; i < styles.length; i++) {
      if (styles[i] && styles[i].hasOwnProperty('fontSize')) {
        fontSizeStyle = styles[i].fontSize;
      }
    }

    expect(fontSizeStyle).toBe(34);
  });

  it('should have font size of 28 when h3', () => {
    const component = shallow(<Text h3 />);

    const styles = component.props().style;
    let fontSizeStyle;
    for (let i = 0; i < styles.length; i++) {
      if (styles[i] && styles[i].hasOwnProperty('fontSize')) {
        fontSizeStyle = styles[i].fontSize;
      }
    }

    expect(fontSizeStyle).toBe(28);
  });

  it('should have font size of 22 when h4', () => {
    const component = shallow(<Text h4 />);

    const styles = component.props().style;
    let fontSizeStyle;
    for (let i = 0; i < styles.length; i++) {
      if (styles[i] && styles[i].hasOwnProperty('fontSize')) {
        fontSizeStyle = styles[i].fontSize;
      }
    }

    expect(fontSizeStyle).toBe(22);
  });

  it('should have text as children', () => {
    const component = shallow(<Text>Children Text</Text>);

    expect(component.props().children).toBe('Children Text');
  });

  it('should render fontFamily and style', () => {
    const component = shallow(
      <Text fontFamily="comic-sans" style={{ color: 'red' }}>
        Children Text
      </Text>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
