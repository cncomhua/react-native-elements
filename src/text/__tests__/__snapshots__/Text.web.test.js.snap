// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misc: Text: Props Misc: Text: [ prop: fontFamily ] enzyme-mount: snapshot: Misc: Text: [ prop: fontFamily ] enzyme-mount: snapshot 1`] = `
<TextElement
  fontFamily="Courier New"
>
  <Text
    style={
      Object {
        "fontFamily": "Courier New",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "fontFamily": "Courier New",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: fontFamily ] enzyme-render: snapshot: Misc: Text: [ prop: fontFamily ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="font-family:Courier New"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: fontFamily ] enzyme-shallow: snapshot: Misc: Text: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "fontFamily": "Courier New",
    }
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h1 ] enzyme-mount: snapshot: Misc: Text: [ prop: h1 ] enzyme-mount: snapshot 1`] = `
<TextElement
  h1={true}
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h1 ] enzyme-render: snapshot: Misc: Text: [ prop: h1 ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="font-size:40px"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h1 ] enzyme-shallow: snapshot: Misc: Text: [ prop: h1 ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 40,
      "fontWeight": "bold",
    }
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h2 ] enzyme-mount: snapshot: Misc: Text: [ prop: h2 ] enzyme-mount: snapshot 1`] = `
<TextElement
  h2={true}
>
  <Text
    style={
      Object {
        "fontSize": 34,
        "fontWeight": "bold",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "fontSize": "34px",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h2 ] enzyme-render: snapshot: Misc: Text: [ prop: h2 ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="font-size:34px"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h2 ] enzyme-shallow: snapshot: Misc: Text: [ prop: h2 ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 34,
      "fontWeight": "bold",
    }
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h3 ] enzyme-mount: snapshot: Misc: Text: [ prop: h3 ] enzyme-mount: snapshot 1`] = `
<TextElement
  h3={true}
>
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "fontSize": "28px",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h3 ] enzyme-render: snapshot: Misc: Text: [ prop: h3 ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="font-size:28px"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h3 ] enzyme-shallow: snapshot: Misc: Text: [ prop: h3 ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 28,
      "fontWeight": "bold",
    }
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h4 ] enzyme-mount: snapshot: Misc: Text: [ prop: h4 ] enzyme-mount: snapshot 1`] = `
<TextElement
  h4={true}
>
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "bold",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "fontSize": "22px",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h4 ] enzyme-render: snapshot: Misc: Text: [ prop: h4 ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontWeight-vw2c0b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="font-size:22px"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: h4 ] enzyme-shallow: snapshot: Misc: Text: [ prop: h4 ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "fontSize": 22,
      "fontWeight": "bold",
    }
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: no props ] enzyme-mount: snapshot: Misc: Text: [ prop: no props ] enzyme-mount: snapshot 1`] = `
<TextElement>
  <Text
    style={
      Array [
        72,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: no props ] enzyme-render: snapshot: Misc: Text: [ prop: no props ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: no props ] enzyme-shallow: snapshot: Misc: Text: [ prop: no props ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Array [
      72,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Text
</Text>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: style ] enzyme-mount: snapshot: Misc: Text: [ prop: style ] enzyme-mount: snapshot 1`] = `
<TextElement
  style={
    Object {
      "color": "#071",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#071",
      }
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "color": "rgba(0,119,17,1)",
        }
      }
    >
      Text
    </div>
  </Text>
</TextElement>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: style ] enzyme-render: snapshot: Misc: Text: [ prop: style ] enzyme-render: snapshot 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="color:rgba(0,119,17,1)"
>
  Text
</div>
`;

exports[`Misc: Text: Props Misc: Text: [ prop: style ] enzyme-shallow: snapshot: Misc: Text: [ prop: style ] enzyme-shallow: snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#071",
    }
  }
>
  Text
</Text>
`;

exports[`Text Component should render fontFamily and style 1`] = `
<Text
  style={
    Object {
      "color": "red",
      "fontFamily": "comic-sans",
    }
  }
>
  Children Text
</Text>
`;

exports[`Text Component should render without issues 1`] = `
<Text
  style={
    Array [
      72,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;
